eng-wasm/src/crypto_wasm.rs

```diff
@@ -0,0 +1,57 @@
+use super::*;
+
+use rand_wasm::Rand;
+use eng_pwasm_abi::types::U256;
+
+const SYMMETRIC_KEY_SIZE: usize = 32;
+const IV_SIZE: usize = 96 / 8;
+const PAYLOAD_SIZE: usize = 1024;
+
+pub type IV = [u8; IV_SIZE];
+pub type SymmetricKey = [u8; SYMMETRIC_KEY_SIZE];
+
+pub fn generate_key() -> SymmetricKey {
+    let key_int: U256 = Rand::gen();
+    let key = H256::from(key_int);
+    key.0
+}
+
+pub fn encrypt(message: &[u8], key: &SymmetricKey) -> Vec<u8> {
+    // TODO: Is this really needed? Dynamically sized buffers don't seem to work.
```

Comment:
As far as I understand, it is impossible to use the dynamic buffers here. You should allocate a continuous region of WASM memory. This region is filled when the runtime function is called. The dynamic array cannot be managed from the runtime.